utils/s2n_key_material.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "utils/s2n_blob.h"
+
+/* The final step in key derivation to use secrets and other information to
+ * perform a key expansion. The key expansion results in an array of contiguous
+ * data which is then interpreted as the mac, key and iv for the client and
+ * server.
+ *
+ * The following is the memory layout of the key information, where 'C' represents
+ * Client and 'S' represents Server.
+ *
+ *     [ C_MAC, S_MAC, C_KEY, S_KEY, C_IV, S_IV ]
+ */
+
+/* Enough to support TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, 2*SHA384_DIGEST_LEN + 2*AES256_KEY_SIZE */
+#define S2N_MAX_KEY_BLOCK_LEN 160
+
+struct s2n_key_material {
+    /* key material data resulting from key expansion */
+    uint8_t data[S2N_MAX_KEY_BLOCK_LEN];
+
+    /* pointers into data representing specific key information */
+    struct s2n_blob client_mac;
+    struct s2n_blob server_mac;
+    struct s2n_blob client_key;
+    struct s2n_blob server_key;
+    struct s2n_blob client_iv;
+    struct s2n_blob server_iv;
+};
+
+/* Types representing specific key information from the key expansion */
+typedef enum {
+    S2N_KEY_MATERIAL_CLIENT_MAC,
+    S2N_KEY_MATERIAL_SERVER_MAC,
+    S2N_KEY_MATERIAL_CLIENT_KEY,
+    S2N_KEY_MATERIAL_SERVER_KEY,
+    S2N_KEY_MATERIAL_CLIENT_IV,
+    S2N_KEY_MATERIAL_SERVER_IV,
+} s2n_key_material_type;
+
+S2N_RESULT s2n_key_material_init(struct s2n_key_material *key_material, struct s2n_connection *conn);
+S2N_RESULT s2n_key_material_read(struct s2n_key_material *key_material, s2n_key_material_type type, struct s2n_blob *out);
```

Comment:
s2n_key_material_read + s2n_key_material_type are unnecessary. They can be replaced by code like `&key_material->client_mac`

This doesn't need to be its own module and can keep living in s2n_prf